privacyidea/lib/framework.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+#  2018-11-15   Friedrich Weber <friedrich.weber@netknights.it>
+#               Add a framework module to reduce the coupling to flask
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+from flask import current_app, g
+# We import the gettext function here and export it as ``_``.
+from flask_babel import gettext as _
+
+
+def get_application_local_store():
```

Comment:
I like the overall concept. 
I should think a bit about the naming. I think the module name "framework" is nice.
These names seem a bit long to me "get_application_local_store()". Should be use the property decorator and shorten it .

I think would like to drop the "get_"-part. But then the call would be

~~~python
appstore = application_local_store()
~~~~
which migt also look a bit wired. Unfortunately the property decorator does not work on functions, only on methods.

Reply:
Yes, I also found naming quite hard :-)
I agree the name is a bit long. I actually think keeping the ``get_`` prefix would be nice, because  it is consistent with a lot of other ``get_`` functions in the lib level. But we could shorten it anyway: As we also have ``get_app_config``, we could shorten this to ``get_app_local_store``? Or even ``get_app_store`` (don't let Apple read this!) and ``get_request_store``?

Reply:
As a first step, I renamed ``get_application_local_store`` to ``get_app_local_store`` in 
484c508. This is also more consistent with ``get_app_config``.